client/platform/desktop/frontend/components/ViewerLoader.vue

```diff
@@ -38,11 +43,29 @@ export default defineComponent({
     },
   },
   setup(props) {
+    const { prompt } = usePrompt();
     const viewerRef = ref();
     const compoundId = ref(props.id);
     const subTypeList = computed(() => [datasets.value[props.id]?.subType || null]);
     const readonlyMode = computed(() => settings.value?.readonlyMode || false);
 
+    watch(runningJobs, async (_previous, current) => {
+      const index = current.findIndex((item) => item.job.datasetIds.includes(props.id));
+      if (index !== -1 && current[index] && current[index].job.exitCode !== -1 && current[index].job.jobType === 'pipeline') {
```

Comment:
Use `find` instead of `findIndex` if you don't actually care about the index and you just want the job.  It will make the conditional on 54 less involved.

```
      const job = current.find((item) => item.job.datasetIds.includes(props.id));
      if (job && job.job.jobType === 'pipeline' && job.job.exitCode !== -1) {
```